socs/agent/smurf_recorder.py

```diff
@@ -210,12 +228,16 @@ def _establish_reader_connection(self, timeout=5):
 
         """
         reader = None
-        try:
-            reader = core.G3Reader(self.address,
-                                   timeout=timeout)
-            self.log.info("G3Reader connection established")
-        except RuntimeError:
-            self.log.error("G3Reader could not connect.")
+
+        if (check_port(self.host, self.port)):
```

Comment:
Since `check_port` has `s.setblocking(1)` does this wait here until a connection is made?

The code below here was written to prevent rapid connection attempts when a connection couldn't be made. Do we run into that situation anymore if we wait to connect until `check_port` passes?

Reply:
As described above, I think the connection was being refused immediately even with a non-zero timeout, so this code block is still needed.